### PR TITLE
[New] Select feature in scene layer 

### PR DESCRIPTION
**Pull request by chriswebb09**
## Description

This PR implements **`Select feature in scene layer `** in `Scenes` category.

`swift/issues/5433`

## How To Test

Tap a feature in the SceneView and check to see that it is highlighted. Select another feature and ensure the previous selection is removed when the new selection is highlighted.

## Screenshots

select-features-in-scene-layer

### CODE REVIEW

File: Samples.xcodeproj/project.pbxproj
Comment (yo1995) [inline Samples.xcodeproj/project.pbxproj:1]: A few things with the project file

- See https://github.com/Esri/arcgis-maps-sdk-swift-samples/pull/367/files for an example on what are the changes look like in the .pbxproj file.
- Don't need to add `ArcGIS` framework. Because we added the toolkit package, which depends on the `ArcGIS` package, you can import it without adding it to the project again.
- Please also add the `.swift` files to the "Copy Source Code Files" build phase. We need to display the source code file in the upper right info button page, so the source code files are copied into the bundle.

File: Samples.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -2741,7 +2756,7 @@
 		001C6DDC27FE5CE800D472C2 /* Create .secrets File If It Does Not Exist */ = {
 			isa = PBXShellScriptBuildPhase;
 			alwaysOutOfDate = 1;
-			buildActionMask = 2147483647;
+			buildActionMask = 12;
```
Comment (yo1995) [inline Samples.xcodeproj/project.pbxproj:2759]: Please revert this change.

File: Shared/Samples/Select features in scene layer/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# Select feature in scene
+
+Select feature in a scene layer.
+
+![Screenshot of select feature in scene layer sample](select-features-in-scene.png)
```
Comment (yo1995) [inline Shared/Samples/Select features in scene layer/README.md:5]:
```suggestion
![Screenshot of Select feature in scene layer sample](select-features-in-scene-layer.png)
```

The README and metadata has specific formatting checks. See in `common-samples/wiki/README-Template` repo wiki for an idea.

p.s. we don't put devtopia links in public repo for security reasons, so be careful with that. No sweat if you accidentally included one - just remove it.

File: Shared/Samples/Select features in scene layer/README.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# Select feature in scene
```
Comment (yo1995) [inline Shared/Samples/Select features in scene layer/README.md:1]:
```suggestion
# Select feature in scene layer
```

File: Shared/Samples/Select features in scene layer/README.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+Tap on a feature in the scene.
+
+## How it works
+
+1. Create a `Scene` instance.
+2. Create instances of `SceneViewReader` and `SceneView`.
+3. Asynchronously identify nearby features at the tapped location from the map view using the `SceneViewProxy.identify(layer:screenPoint:tolerance:maximumResults:)` method.
+4. Select all identified features in the feature layer with `ArcGISSceneLayer.select(features:)`.
+
```
Comment (yo1995) [inline Shared/Samples/Select features in scene layer/README.md:21]:
```suggestion
```

File: Shared/Samples/Select features in scene layer/select-features-in-scene.png
Comment (yo1995) [inline Shared/Samples/Select features in scene layer/select-features-in-scene.png:1]: yikes - binary file size 😢 
please read the contribution guideline again to double check if you missed sth
Also, please rename the screenshot in the next commit.

File: Shared/Samples/Select features in scene layer/README.metadata.json
Comment (yo1995) [inline Shared/Samples/Select features in scene layer/README.metadata.json:1]: Next week we can discuss the usage of `Scripts/CI/create-metadata-from-README.py`. It helps you to generate a template for the metadata json.

File: Shared/Samples/Select features in scene layer/README.metadata.json [hunk 5]
```diff
@@ -0,0 +1,24 @@
+{
+    "category": "Scenes",
+    "description": "Identify features in a scene to select.",
+    "ignore": false,
+    "images": ["select-features-in-scene.png"],
```
Comment (yo1995) [inline Shared/Samples/Select features in scene layer/README.metadata.json:5]:
```suggestion
    "images": [
        "select-features-in-scene.png"
    ],
```
We prefer put the filename on its own line so future changes can be limited to each line.

File: Shared/Samples/Select features in scene layer/SelectFeaturesInSceneLayerView.swift [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct SelectFeaturesInSceneLayerView: View {
+    
+    // The point on the screen where the user tapped.
+    //
+    @State private var tapPoint: CGPoint?
```
Comment (yo1995) [inline Shared/Samples/Select features in scene layer/SelectFeaturesInSceneLayerView.swift:22]: See other samples for the docstring styling. Use `cmd` `option` `/` to let Xcode generate a stub for comment or docstring.

**Review by yo1995 — COMMENTED** [review]
Some housekeeping suggestions.

File: Shared/Samples/Select features in scene layer/select-features-in-scene.png
Comment (chriswebb09) [reply Shared/Samples/Select features in scene layer/select-features-in-scene.png:1]: it's at 60kb now

File: Shared/Samples/Select features in scene layer/README.md (hunk 4 above)
Comment (chriswebb09) [reply Shared/Samples/Select features in scene layer/README.md:21]: There is only one line spacing now

File: Shared/Samples/Select features in scene layer/SelectFeaturesInSceneLayerView.swift [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+                }
+                .task(id: tapPoint) {
+                    // clear previous selections
+                    sceneLayer.clearSelection()
+                    do {
+                        // ensure there is a tap point captures
+                        guard let locationPoint = tapPoint else { return }
+                        // get a list of possible of the elements that were tapped
+                        let result = try await sceneViewProxy.identify(on: sceneLayer, screenPoint: locationPoint, tolerance: 10)
+                        // grab the first selected element in the array
```
Comment (yo1995) [inline Shared/Samples/Select features in scene layer/SelectFeaturesInSceneLayerView.swift:65]:
```suggestion
                        // Gets the first feature from the identify results.
```

File: Shared/Samples/Select features in scene layer/SelectFeaturesInSceneLayerView.swift [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+                        guard let feature = result.geoElements.first as? ArcGISFeature else { return }
+                        // highlight the selected feature
```
Comment (yo1995) [inline Shared/Samples/Select features in scene layer/SelectFeaturesInSceneLayerView.swift:67]:
```suggestion
                        // Selects the feature in the scene layer.
```

File: Shared/Samples/Select features in scene layer/SelectFeaturesInSceneLayerView.swift [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+                        sceneLayer.selectFeature(feature)
+                    } catch {
+                       return
+                    }
```
Comment (yo1995) [inline Shared/Samples/Select features in scene layer/SelectFeaturesInSceneLayerView.swift:71]: See https://github.com/Esri/arcgis-maps-sdk-swift-samples/blob/main/Shared/Samples/Select%20features%20in%20feature%20layer/SelectFeaturesInFeatureLayerView.swift as an example on how errors are handled.

File: Shared/Samples/Select features in scene layer/SelectFeaturesInSceneLayerView.swift [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+        return scene
+    }()
+    
+    /// Add feature layer to scene in initialization
+    init() {
+        scene.addOperationalLayer(sceneLayer)
+    }
+    
+    var body: some View {
+        // Wrap the SceneView in SceneViewReader and expose its SceneViewProxy.
```
[the remark this answers is not in the document]
Comment (chriswebb09) [reply Shared/Samples/Select features in scene layer/SelectFeaturesInSceneLayerView.swift:50]: Would it be appropriate to add this link in the comments along with the brief explanation?  https://community.esri.com/t5/arcgis-maps-sdks-native-blog/let-s-build-with-swift-4-view-modifier-and-reader/ba-p/1294852

File: Shared/Samples/Select features in scene layer/SelectFeaturesInSceneLayerView.swift [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+        surface.addElevationSource(ArcGISTiledElevationSource(url: .worldElevationService))
+        // Sets the surface to the scene's base surface.
+        scene.baseSurface = surface
+        return scene
+    }()
+    
+    /// Add feature layer to scene in initialization
+    init() {
+        scene.addOperationalLayer(sceneLayer)
+    }
```
[the remark this answers is not in the document]
Comment (yo1995) [reply Shared/Samples/Select features in scene layer/SelectFeaturesInSceneLayerView.swift:47]: Sorry I didn't make myself clear. What you have here is totally fine, no need to change. 
Another approach can be adding a computed property to get the layer from the array instead of holding it as a state property.

```swift
    private var sceneLayer: ArcGISSceneLayer {
        scene.operationalLayers.first as! ArcGISSceneLayer
    }
```

File: Shared/Samples/Select features in scene layer/SelectFeaturesInSceneLayerView.swift (hunk 10 above)
[the remark this answers is not in the document]
Comment (yo1995) [reply Shared/Samples/Select features in scene layer/SelectFeaturesInSceneLayerView.swift:50]: That can be a bit too much. We rely on the ArcGIS Developers website for detailed explanation and mostly don't put links in Samples comments. Comments are better to be succinct while touch the key concepts.

File: Shared/Samples/Select features in scene layer/README.md [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+## How to use the sample
+
+Tap on a feature in the scene.
+
+## How it works
+
+1. Create a `Scene` instance.
+2. Create instances of `SceneViewReader` and `SceneView`.
+3. Asynchronously identify nearby features at the tapped location from the map view using the `SceneViewProxy.identify(layer:screenPoint:tolerance:maximumResults:)` method.
+4. Select all identified features in the feature layer with `ArcGISSceneLayer.select(features:)`.
```
[the remark this answers is not in the document]
Comment (chriswebb09) [reply Shared/Samples/Select features in scene layer/README.md]: Gotcha, I think I got it as close to the template. If it looks good, I'll resolve this.

File: Shared/Samples/Select features in scene layer/SelectFeaturesInSceneLayerView.swift (hunk 9 above)
Comment (chriswebb09) [reply Shared/Samples/Select features in scene layer/SelectFeaturesInSceneLayerView.swift]: Gotcha, I think it handles correctly. When tested with random error it shows error popup now.

**Review by CalebRas — COMMENTED** [review]
Hi Chris! Thanks for implementing this sample. Just have a few suggestions for you:

**Review by yo1995 — DISMISSED** [review]
Great work on your first sample! 🎉

**Review by CalebRas — DISMISSED** [review]
Looks good! Great job Chris

**Review by yo1995 — APPROVED** [review]

**Review by CalebRas — APPROVED** [review]
